app/scraper/sources/face_book.py

```diff
@@ -133,108 +134,114 @@ def scrape(self, current_people):
+                if directory_entry is not None:
+                    person = self.directory.merge_one(person, directory_entry)
                 else:
-                    logger.info('Image has not been processed yet.')
-                    image_r = requests.get('https://students.yale.edu/facebook/Photo?id=' + str(image_id),
-                                           headers={'Cookie': self.cookie},
-                                           stream=True)
-                    image_r.raw.decode_content = True
-                    try:
-                        im = Image.open(image_r.raw)
-
-                        # Paste mask over watermark
-                        im.paste(watermark_mask, (0, 0), watermark_mask)
-
-                        output = BytesIO()
-                        im.save(output, format='JPEG', mode='RGB')
-
-                        person['image'] = self.image_uploader.upload_image(output, image_filename)
-                    except OSError:
-                        # "Cannot identify image" error
-                        logger.info('PIL could not identify image.')
-
-            if person.get('email'):
-                emails[person['email']] = len(people)
-            people.append(person)
+                    logger.info('Could not find directory entry.')
+
+                image_id = self.clean_image_id(container.find('img')['src'])
+                if image_id:
+                    image_filename = self.image_uploader.get_image_filename(image_id, person)
+                    if image_filename in self.image_uploader.files:
+                        person['image'] = self.image_uploader.get_file_url(image_filename)
+                    else:
+                        logger.info('Image has not been processed yet.')
+                        image_r = requests.get('https://students.yale.edu/facebook/Photo?id=' + str(image_id),
+                                            headers={'Cookie': self.cookie},
+                                            stream=True)
+                        image_r.raw.decode_content = True
+                        try:
+                            im = Image.open(image_r.raw)
+
+                            # Paste mask over watermark
+                            im.paste(watermark_mask, (0, 0), watermark_mask)
+
+                            output = BytesIO()
+                            im.save(output, format='JPEG', mode='RGB')
+
+                            person['image'] = self.image_uploader.upload_image(output, image_filename)
+                        except OSError:
+                            # "Cannot identify image" error
+                            logger.info('PIL could not identify image.')
+
+                if person.get('email'):
+                    emails[person['email']] = len(people)
+                people.append(person)
+            except Exception as e:
+                logger.error('Error parsing person.')
+                logger.error(e)
```

Comment:
Wait, I don't think this addresses the root cause. We need to actually stop execution and propagate the error up to the main thread. Logging of the error is already happening.